### ISSUE TITLE
Document Bearer Authorization header

### ISSUE THREAD

**Issue by manuio** [Mainflux/mainflux-lite#1565]
**FEATURE REQUEST**

2. Describe the feature you are requesting, as well as the possible use case(s) for it.
Document Bearer Authentication header. This is related to #1523 documentation.

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).
must have

**Comment by manuio** [comment]
@drasko @dusanb94 @mteodor @blokovi I need your opinion about Auth security schemes: https://swagger.io/docs/specification/authentication/
Looks like we were using the Auth header as parameter while it's a securityScheme. The problem here is that only some HTTP schemes are valid. Should wer use Bearer for User token and Basic for Thing key?

**Comment by drasko** [comment]
@manuio I am not sure I understand the question. AFAIK, @mteodor introduced `Bearer` for user-initiated API calls and `Thing` for device-initiated API calls.

**Comment by manuio** [comment]
@drasko Right, but `thing` cannot be used as scheme type. I decided to define both of them as bearer with different description.

**Comment by drasko** [comment]
I still do not understand - both of what? What scheme do you mention? Can you please add some reference links and a longer explanation, I do not have all details.

**Comment by manuio** [comment]
@drasko openapi auth scheme. Check the link above.

**Comment by drasko** [comment]
@manuio I see what you mean - no, we should use `Thing`, not `Basic`. Using custom scheme is not a problem.

We know that `Thing` scheme is not by standard, but it is not unusual to use custom schemes (even Amazon is doing this), and @mteodor and @dusanb94 opted for this option.

Both user and device in our case use `Bearer` scheme strictum sensum, and I this is why I wanted that we somehow encode this information in the token itself. However, this is not quite possible, as user uses JWT and Thing uses simplestring, so without different scheme flag we would not even know how to decode the tolen to check it's type.

**Comment by drasko** [comment]
BTW, please take a look at the discussion [here](https://github.com/mainflux/mainflux/pull/1470#discussion_r786319670). You will see for example how AWS ([here](https://www.loginradius.com/blog/async/everything-you-want-to-know-about-authorization-headers/)) introduces custom auth scheme tags. And so on.

**Comment by manuio** [comment]
@drasko I'm aware of this discussion and I understand that we are using a custom scheme. The thing is that I was confused by our implementation of Basic scheme in http-adapter. After to verify I saw that there we extract the thing key from user password. It's a kind of hack but it's not simply a different prefix as I thought. 

Let me resume what I did. I'm using `bearer` scheme everywhere with `Bearer` or `Thing` prefix depending on if we are sending a JWT user token or and UUID Thing key. For http-adapter Thing and Basic schemes are accepted. For readers it's Bearer or Thing. For Bootstrap depending onm the endpoint it's Bearer or Thing.

In fact my question was more about openapi since custom types are not accepted. what I did is to use `bearer` type everywhere and explain the  type in the description.
